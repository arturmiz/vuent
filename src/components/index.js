import './Vnt.vue';
export { default as VntAutosuggest } from './autosuggest';
export { default as VntButton } from './button';
export { default as VntCheckbox } from './checkbox';
export { default as VntDialog } from './dialog';
export { default as VntHeader } from './header';
export { default as VntIcon } from './icon';
export { default as VntInput } from './input';
export { default as VntLink } from './link';
export { default as VntListView } from './listview';
export { default as VntNavView } from './navview';
export { default as VntPivot } from './pivot';
export { default as VntRadio } from './radio';
export { default as VntRating } from './rating';
export { default as VntSelect } from './select';
export { default as VntSlider } from './slider';
export { default as VntToggle } from './toggle';
