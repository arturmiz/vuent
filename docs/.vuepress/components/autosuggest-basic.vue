<template>
  <div class="playground">
    <vnt-autosuggest v-model="country"
                     :options="countries"></vnt-autosuggest>
    <div class="result">country: {{ country }}</div>
  </div>
</template>

<script>
import { VntAutosuggest } from '../../../src/components';

export default {
  name: 'AutosuggestBasic',
  components: {
    VntAutosuggest
  },
  data() {
    return {
      country: null,
      countries: [
        'Australia',
        'Belgium',
        'Canada',
        'China',
        'France',
        'Germany',
        'Poland',
        'United Kingdom',
        'United States',
      ]
    }
  }
}
</script>
