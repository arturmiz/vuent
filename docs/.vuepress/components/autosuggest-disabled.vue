<template>
  <div class="playground">
    <vnt-autosuggest v-model="country"
                     :options="countries" disabled></vnt-autosuggest>
    <div class="result">country: {{ country }}</div>
  </div>
</template>

<script>
import { VntAutosuggest } from '../../../src/components';

export default {
  name: 'AutosuggestDisabled',
  components: {
    VntAutosuggest
  },
  data() {
    return {
      country: null,
      countries: [
        'Australia',
        'Belgium',
        'Canada',
        'China',
        'France',
        'Germany',
        'Poland',
        'United Kingdom',
        'United States',
      ]
    }
  }
}
</script>
