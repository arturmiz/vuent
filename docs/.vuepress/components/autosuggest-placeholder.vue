<template>
  <div class="playground">
    <vnt-autosuggest v-model="country"
                     :options="countries"
                     placeholder="Type something..."></vnt-autosuggest>
    <div class="result">country: {{ country }}</div>
  </div>
</template>

<script>
import { VntAutosuggest } from '../../../src/components';

export default {
  name: 'AutosuggestPlaceholder',
  components: {
    VntAutosuggest
  },
  data() {
    return {
      country: null,
      countries: [
        'Australia',
        'Belgium',
        'Canada',
        'China',
        'France',
        'Germany',
        'Poland',
        'United Kingdom',
        'United States',
      ]
    }
  }
}
</script>
