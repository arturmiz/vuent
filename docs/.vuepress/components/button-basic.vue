<template>
  <div class="playground">
    <vnt-button>Button</vnt-button>
  </div>
</template>

<script>
import { VntButton } from '../../../src/components';

export default {
  name: 'ButtonBasic',
  components: {
    VntButton
  }
}
</script>
