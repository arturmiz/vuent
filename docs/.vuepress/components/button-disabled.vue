<template>
  <div class="playground">
    <vnt-button disabled>Button</vnt-button>
  </div>
</template>

<script>
import { VntButton } from '../../../src/components';

export default {
  name: 'ButtonDisabled',
  components: {
    VntButton
  }
}
</script>
