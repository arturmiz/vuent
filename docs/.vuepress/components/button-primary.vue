<template>
  <div class="playground">
    <vnt-button primary>Primary</vnt-button>
  </div>
</template>

<script>
import { VntButton } from '../../../src/components';

export default {
  name: 'ButtonPrimary',
  components: {
    VntButton
  }
}
</script>
