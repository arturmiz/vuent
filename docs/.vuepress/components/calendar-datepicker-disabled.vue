<template>
  <div class="playground">
    <vnt-calendar-datepicker v-model="date" disabled />
    <div class="result">date: {{ date }}</div>
  </div>
</template>

<script>
import { VntCalendarDatepicker } from "../../../src/components";

export default {
  name: "CalendarDatepickerBasic",
  components: {
    VntCalendarDatepicker
  },
  data() {
    return {
      date: "2020-04-05"
    };
  }
};
</script>
