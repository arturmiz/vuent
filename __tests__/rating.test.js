import { createLocalVue, mount } from '@vue/test-utils';
import { VntRating } from '@/components';
import { isInstalled } from './utils';

describe('Rating', () => {
  let wrapper;

  test('can be installed separately', () => {
    const localVue = createLocalVue();
    localVue.use(VntRating);

    expect(isInstalled(localVue, VntRating)).toBe(true);
  });

  describe('by default', () => {

    beforeAll(() => {
      wrapper = mount(VntRating);
    });

    test('is not disabled', () => {
      expect(wrapper.vm.disabled).toBe(false);
    });

    test('stars number is 5', () => {
      expect(wrapper.vm.max).toBe(5);
    });

    test('value is 0', () => {
      expect(wrapper.vm.value).toBe(0);
    });

    test('renders correctly', () => {
      expect(wrapper.html()).toMatchSnapshot();
    });

  });

  describe('when not 0 value provided', () => {

    beforeAll(() => {
      wrapper = mount(VntRating, {
        propsData: {
          value: 2.64
        }
      });
    });

    test('calculated value is 2', () => {
      expect(wrapper.vm.starsValue).toBe(2);
    });

    test('renders correctly', () => {
      expect(wrapper.html()).toMatchSnapshot();
    });

  });

  describe('can be disabled', () => {

    beforeAll(() => {
      wrapper = mount(VntRating, {
        propsData: {
          disabled: true,
          value: 3
        }
      });
    });

    test('is disabled', () => {
      expect(wrapper.vm.disabled).toBe(true);
    });

    test('renders correctly', () => {
      expect(wrapper.html()).toMatchSnapshot();
    });

  });

  describe('when not disabled, updates rating', () => {

    beforeAll(() => {
      wrapper = mount(VntRating, {
        propsData: {
          value: 2
        }
      });

      const label = wrapper.find('label[aria-label="3"]');
      label.trigger('click');
    });

    test('event is emitted', () => {
      expect(wrapper.emitted()['change']).toHaveLength(1);
    });

    test('new rating value is emitted', () => {
      const [newRating] = wrapper.emitted()['change'];
      expect(newRating).toEqual([3]);
    });

  });

  describe('when disabled, doesn\'t change rating', () => {

    beforeAll(() => {
      wrapper = mount(VntRating, {
        propsData: {
          value: 2,
          disabled: true
        }
      });

      const label = wrapper.find('label[aria-label="3"]');
      label.trigger('click');
    });

    test('event is not emitted', () => {
      expect(wrapper.emitted()['change']).toBeUndefined();
    });

  });

});
