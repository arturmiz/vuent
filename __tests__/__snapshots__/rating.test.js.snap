// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rating by default renders correctly 1`] = `
<div tabindex="0" aria-label="0" class="vnt-rating">
  <label aria-label="5" tabindex="0" class="vnt-rating__star vnt-rating__star--off">★</label>
  <label aria-label="4" tabindex="0" class="vnt-rating__star vnt-rating__star--off">★</label>
  <label aria-label="3" tabindex="0" class="vnt-rating__star vnt-rating__star--off">★</label>
  <label aria-label="2" tabindex="0" class="vnt-rating__star vnt-rating__star--off">★</label>
  <label aria-label="1" tabindex="0" class="vnt-rating__star vnt-rating__star--off">★</label>
</div>
`;

exports[`Rating can be disabled renders correctly 1`] = `
<div tabindex="0" aria-disabled="true" aria-label="3" class="vnt-rating vnt-rating--disabled">
  <label aria-disabled="true" aria-label="5" tabindex="0" class="vnt-rating__star vnt-rating__star--off">★</label>
  <label aria-disabled="true" aria-label="4" tabindex="0" class="vnt-rating__star vnt-rating__star--off">★</label>
  <label aria-disabled="true" aria-label="3" tabindex="0" class="vnt-rating__star">★</label>
  <label aria-disabled="true" aria-label="2" tabindex="0" class="vnt-rating__star">★</label>
  <label aria-disabled="true" aria-label="1" tabindex="0" class="vnt-rating__star">★</label>
</div>
`;

exports[`Rating when not 0 value provided renders correctly 1`] = `
<div tabindex="0" aria-label="2.64" class="vnt-rating">
  <label aria-label="5" tabindex="0" class="vnt-rating__star vnt-rating__star--off">★</label>
  <label aria-label="4" tabindex="0" class="vnt-rating__star vnt-rating__star--off">★</label>
  <label aria-label="3" tabindex="0" class="vnt-rating__star vnt-rating__star--off">★</label>
  <label aria-label="2" tabindex="0" class="vnt-rating__star">★</label>
  <label aria-label="1" tabindex="0" class="vnt-rating__star">★</label>
</div>
`;
